numba/core/typing/new_builtins.py

```diff
@@ -0,0 +1,1178 @@
+        implementation.  Specifically, the lack of a higher-order type
+        (i.e. passing the ``DictType`` vs ``DictType(key_type, value_type)``)
+        """
+        ty = classty.instance_type
+
+        if isinstance(ty, type) and issubclass(ty, types.Type):
+            # Redirect the typing to a:
+            #   @type_callable(ty)
+            #   def typeddict_call(context):
+            #        ...
+            # For example, see numba/typed/typeddict.py
+            #   @type_callable(DictType)
+            #   def typeddict_call(context):
+            class Redirect(object):
+
+                def __init__(self, context):
+                    self.context =  context
+
+                def __call__(self, *args, **kwargs):
+                    result = self.context.resolve_function_type(ty, args, kwargs)
+                    if hasattr(result, "pysig"):
+                        self.pysig = result.pysig
+                    return result
+
+            return types.Function(make_callable_template(key=ty,
+                                                         typer=Redirect(self.context)))
+
+
+#------------------------------------------------------------------------------
+
+
+class MinMaxBase(AbstractTemplate):
+
+    def _unify_minmax(self, tys):
+        for ty in tys:
+            if not isinstance(ty, (types.Number, types.NPDatetime, types.NPTimedelta)):
+                return
+        return self.context.unify_types(*tys)
+
+    def generic(self, args, kws):
+        """
+        Resolve a min() or max() call.
+        """
+        assert not kws
+
+        if not args:
+            return
+        if len(args) == 1:
+            # max(arg) only supported if arg is an iterable
+            if isinstance(args[0], types.BaseTuple):
+                tys = list(args[0])
+                if not tys:
+                    raise TypeError("%s() argument is an empty tuple"
+                                    % (self.key.__name__,))
+            else:
+                return
+        else:
+            # max(*args)
+            tys = args
+        retty = self._unify_minmax(tys)
+        if retty is not None:
+            return signature(retty, *args)
+
+
+@infer_global(max)
+class Max(MinMaxBase):
+    pass
+
+
+@infer_global(min)
+class Min(MinMaxBase):
+    pass
+
+
+@infer_global(round)
+class Round(ConcreteTemplate):
+    cases = [
+        signature(types.np_intp, types.np_float32),
+        signature(types.np_int64, types.np_float64),
+        signature(types.np_float32, types.np_float32, types.np_intp),
+        signature(types.np_float64, types.np_float64, types.np_intp),
+    ]
+
+
+#------------------------------------------------------------------------------
+
+
+@infer_global(bool)
+class Bool(AbstractTemplate):
+
+    def generic(self, args, kws):
+        assert not kws
+        [arg] = args
+        if isinstance(arg, (types.Boolean, types.Number)):
+            return signature(types.np_bool_, arg)
```

Comment:
```suggestion
            return signature(types.py_bool, arg)
```
boolean constructor `bool` should return a Python bool?